modules/idle-scheduler/src/lib.rs

```diff
@@ -76,14 +86,56 @@ pub mod module {
 	#[pallet::getter(fn next_task_id)]
 	pub type NextTaskId<T: Config> = StorageValue<_, Nonce, ValueQuery>;
 
+	///
+	#[pallet::storage]
+	#[pallet::getter(fn previous_relay_block)]
+	pub type PreviousRelayBlockNumber<T: Config> = StorageValue<_, BlockNumber, ValueQuery>;
+
 	#[pallet::pallet]
 	#[pallet::without_storage_info]
 	pub struct Pallet<T>(_);
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(_n: T::BlockNumber) -> Weight {
+			// `try_into()` shouldn't fail, if it does, storage will be left empty which will return a value of
+			// zero as default when read. This is the previous relay block because `on_initialize` is executed
+			// before the inherent that sets the new relay chain block number
+			let previous_relay_block: Option<BlockNumber> =
```

Comment:
we should have a `type BlockNumberOf<T>` using the type from `BlockNumberProvider` so we don't need to deal with 3 different (but the same) block number types here.